### ISSUE TITLE
Add `.current` method back to ShopifyAPI::Shop and ShopifyAPI::RecurringApplicationCharge

### ISSUE THREAD

**Issue by scart88** [Shopify/shopify_api#923]
# Issue summary

With the new breaking api change, `ShopifyAPI::Shop.current`, `ShopifyAPI::RecurringApplicationCharge.current` are missing. 
Is there any plan in adding these back?

## Expected behavior

`ShopifyAPI::Shop.current` should return the current shop from the session. 

## Actual behavior

Right now it returns 
`NoMethodError: undefined method 'current' for ShopifyAPI::Shop:Class`

## Steps to reproduce the problem

```
Shop.first.with_shopify_session do
  puts ShopifyAPI::Shop.current
end 
```

## Specifications

- `shopify_api` version: 10.0.0
- Shopify API version used (example: `2022-04`): 2022-04

**Comment by mllemango** [comment]
We're adding this to the TODO list! Keeping this issue open to let others also chime in their feedback

**Comment by stuartchaney** [comment]
+1, this was the last hurdle that prevented us from upgrading version :(

**Comment by marisveide** [comment]
+1 from me as well.

**Comment by jaredbeck** [comment]
I, also, do not see any mention of this in the changelog. Will the changelog be updated?

Until this method is restored, what is the workaround?

**Comment by asecondwill** [comment]
@axlekb  @mllemango  - does this workaround really work reliably?  It isnt very descriptive.  At first glance it could just get the first shop that has this app or something.  

Like the others above, I used the  ```` ShopifyAPI::Shop.current  ````  method a lot.  Upgrading to shopify_app 20+ and so shopify API 10+ is being presented as mandatory by the approvals team, including delisting our app and saying we have to update to be re-evaluated.    So it's pretty important this stuff is either put right or workarounds / changes properly documented.

**Comment by sativva** [comment]
I'm using that instead, waiting for upgrase. xo

`ShopifyAPI::Shop.all(session: @session).first`

**Comment by immakdas** [comment]
Sometimes I think the dev team in Shopify has no idea what they are doing. Instead of improving, things became harder to handle. Current [app](https://github.com/Shopify/shopify_app) version is v21 but you still have to write tons of custom code to handle payments, trial/ decline payments etc

**Comment by sagar-ranglani** [comment]
> This works: `ShopifyAPI::Shop.all[0]`

Does this really work?

**Comment by kirillplatonov** [comment]
The old method was very convenient. Not sure what was the motivation for removing it during the rewrite. For now, I added this to my Rails initializer to restore old behavior:
```rb
# config/initializers/shopify_app.rb

Rails.application.config.after_initialize do
  class ShopifyAPI::Shop
    def self.current(session: ShopifyAPI::Context.active_session, fields: nil)
      ShopifyAPI::Shop.all(session: session, fields: fields).first
    end
  end
end
```

**Comment by sagar-ranglani** [comment]
Any updates on this?

**Comment by jefflewis** [comment]
This is still a major missing feature that we're all having to add to our codebases.

When this library was rewritten, the V10 changelog says this:

>We've rewritten this library for v10, so that it provides all essential features for a Shopify app without depending on the [Active Resource](https://github.com/rails/activeresource) or [graphql-client](https://github.com/github/graphql-client) libraries.

This issue is glaring proof that "all essential features" are not yet provided. This version should have stayed a "beta" release or something like it until a minimum amount of feature parity was in place. The "upgrade" of this gem in order to meet with Shopify's required changes feels like it has been done with very little consideration of the users of this library.

In short, here's my 👍  for adding this (and so many similar) convenience features back to this library so I an get rid of all these monkey patches I'm adding in the gem initializer.

**Comment by scart88** [comment]
Any updates on this?

**Comment by kirillplatonov** [comment]
Not stale

**Comment by Icmus23** [comment]
I'm currently using this approach:
```ruby
client = ShopifyAPI::Clients::Rest::Admin.new(session: @session)
client.get(path: 'shop').body['shop']
```

**Comment by jefflewis** [comment]
Not stale.

**Comment by djkgamc** [comment]
annoying

**Comment by codev009** [comment]
> The old method was very convenient. Not sure what was the motivation for removing it during the rewrite. For now, I added this to my Rails initializer to restore old behavior:
> 
> ```ruby
> # config/initializers/shopify_app.rb
> 
> Rails.application.config.after_initialize do
>   class ShopifyAPI::Shop
>     def self.current(session: ShopifyAPI::Context.active_session, fields: nil)
>       ShopifyAPI::Shop.all(session: session, fields: fields).first
>     end
>   end
> end
> ```

I'm using the latest api version.
This is not working for me. Has anyone solved this yet?
